### PR TITLE
WIP Fix kafka it

### LINKED ISSUE #1011: Fix Kafka storage tests
Kafka storage tests randomly fails https://travis-ci.org/jaegertracing/jaeger/jobs/418644623

```
-- FAIL: TestKafkaStorage (6.21s)
    --- FAIL: TestKafkaStorage/GetTrace (3.02s)
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 1 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 2 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 3 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 4 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 5 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 6 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 7 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 8 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 9 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 10 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 11 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 12 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 13 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 14 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 15 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 16 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 17 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 18 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 19 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 20 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 21 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 22 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 23 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 24 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 25 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 26 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 27 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 28 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 29 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:99: Waiting for storage backend to update documents, iteration 30 out of 30
    	integration_test.go:180: Trace was not found
    	integration_test.go:180: Trace was not found
    	assertions.go:256: 
    			Error Trace:	integration_test.go:184
    			            				integration_test.go:377
    			Error:      	Should be true
    			Test:       	TestKafkaStorage/GetTrace
    	assertions.go:256: 
    			Error Trace:	domain_trace_compare_test.go:50
    			            				integration_test.go:185
    			            				integration_test.go:377
    			Error:      	Expected value not to be nil.
    			Test:       	TestKafkaStorage/GetTrace
```

### PR DESCRIPTION
**Pull request by pavolloffay**
Resolves #1011

### CODE REVIEW

File: scripts/travis/kafka-integration-test.sh [hunk 1]
```diff
@@ -2,16 +2,19 @@
 
 set -e
 
-docker pull spotify/kafka
-CID=$(docker run -d -p 2181:2181 -p 9092:9092 --env ADVERTISED_HOST=localhost --env ADVERTISED_PORT=9092 --rm spotify/kafka)
+docker pull wurstmeister/kafka
```
Comment (pavolloffay) [inline scripts/travis/kafka-integration-test.sh]: note that this is using kafka 2.x before it was 1.x.

`wurstmeister/kafka` seems to be an active project. The `spotify/kafka` seems to be unmaintained.

Comment (yurishkuro) [reply scripts/travis/kafka-integration-test.sh]: Shouldn't we use the one from confluence? That one is definitely supported.

Comment (objectiser) [reply scripts/travis/kafka-integration-test.sh]: There is also http://strimzi.io which is maintained by Red Hat :)

**Comment by pavolloffay** [comment]
Thanks for approving but it seems that CI is still failing

File: scripts/travis/kafka-integration-test.sh (hunk 1 above)
Comment (pavolloffay) [reply scripts/travis/kafka-integration-test.sh]: @yurishkuro which one do you mean?  This docker pull confluentinc/cp-kafka? it seems it's packaged with more confluent specific stuff.

One thing I don't like about these other images that they follow their versioning which is different to kafka versions.

**Comment by pavolloffay** [comment]
#1068 should solve the problem with failing test. I will use this PR to change image distribution. Spotify kafka image does not seems to be maintained, and there is only one version available https://kafka.apache.org/downloads#0.10.0.1 which is quite old.

Do we have any requirements on kafka version? Or just use the same version as is supported by sarama? https://github.com/Shopify/sarama#compatibility-and-api-stability See against what version it is tested https://github.com/Shopify/sarama/blob/master/.travis.yml#L14

**Comment by vprithvi** [comment]
@pavolloffay We don't have any requirements on Kafka versions apart from the compatibility with Sarama. I feel that we should update to the most popular Kafka release, which is probably 0.11, or 1.xx.

**Comment by yurishkuro** [comment]
@pavolloffay is this still an issue? It looks like the builds have been stable recently.

**Comment by pavolloffay** [comment]
I will close it for now. I think we should update kafka version, but that should be done in a separate PR

**Pull request closed without merging**